### ISSUE TITLE
Question about image size

### ISSUE THREAD

**Issue by Sanster** [DQiaole/ZITS_inpainting#7]
Thank you for sharing your great works.

I am planning to integrate your pre-train model into [lama-cleaner](https://github.com/Sanster/lama-cleaner). However, I noticed that the `single_image_test.py` script only supports square images, is this a limitation in the network structure? Or is it just a limitation of this test script?

<img width="836" alt="截屏2022-03-28 下午5 54 42" src="https://user-images.githubusercontent.com/3998421/160373802-1121bf40-1085-43ad-8866-0d779fb9792b.png">

**Comment by DQiaole** [comment]
Hi~

Our TSR only supports the input and output of squares. If you want to test other images, you can resize the image to the approximate square size and then resize back, there should be no large performance impact on images close to squares. For extreme sizes, we cannot ensure the performance.

**Comment by Sanster** [comment]
Thank you for your reply.

Share some square images test result on both LaMa and ZITS:

|Original|Mask|LaMa| This Paper | 2022-03-20 update TSR |
|--------|------|------| -----------| --- |
|![6458524847_2f4c361183_k](https://user-images.githubusercontent.com/3998421/160757560-c6710695-b3f1-454e-b4cb-ce2485d58b35.png)|![6458524847_2f4c361183_k_mask](https://user-images.githubusercontent.com/3998421/160757571-41cb1b00-64f0-438b-9a03-8a53fe571dc6.png)|![6458524847_2f4c361183_k_lama_result](https://user-images.githubusercontent.com/3998421/160757588-9c428365-a8c4-4cb7-a997-74dc2b57a737.jpeg)|![6458524847_2f4c361183_k_result](https://user-images.githubusercontent.com/3998421/160757619-95cd3ddf-35b0-4367-acac-3987a3babc71.png)|![6458524847_2f4c361183_k_result_new](https://user-images.githubusercontent.com/3998421/160804864-030494c6-3de7-4053-8475-aa3dc8e7ff6d.png)
|![6458524847_2f4c361183_k_keep_ratio](https://user-images.githubusercontent.com/3998421/160757686-bd9a2708-8962-4fba-a62e-886542659aaa.png)|![6458524847_2f4c361183_k_keep_ratio_mask](https://user-images.githubusercontent.com/3998421/160757694-aff2d039-f9bd-4937-bc05-0f96f48abba9.png)|![6458524847_2f4c361183_k_keep_ratio_lama_result](https://user-images.githubusercontent.com/3998421/160757712-00527338-52af-4ac5-ad72-84e2677c739e.png)|![6458524847_2f4c361183_k_keep_ratio_result](https://user-images.githubusercontent.com/3998421/160757752-649b76d1-1736-4373-af5f-af6142c62e2d.png)|![6458524847_2f4c361183_k_keep_ratio_result_new](https://user-images.githubusercontent.com/3998421/160805026-352115db-eb03-45a5-8310-77899a0c9bdf.png)|
|![alex-iby-G_Pk4D9rMLs](https://user-images.githubusercontent.com/3998421/160757797-0992b637-3647-477b-b07b-43e990f22491.png)|![alex-iby-G_Pk4D9rMLs_mask](https://user-images.githubusercontent.com/3998421/160757810-573dbd73-3927-4969-a5ec-0ef78302a582.png)|![alex-iby-G_Pk4D9rMLs_lama](https://user-images.githubusercontent.com/3998421/160757827-5c67f59d-e6ae-44d2-bf45-0effae73827f.png)|![alex-iby-G_Pk4D9rMLs_result](https://user-images.githubusercontent.com/3998421/160757852-0ec1751d-39ed-40f9-9a16-82c2571a596e.png)|![alex-iby-G_Pk4D9rMLs_result_new](https://user-images.githubusercontent.com/3998421/160805184-b3e11892-c0ca-41fb-aee6-1a988a41db85.png)|

**Comment by ewrfcas** [comment]
Hi, thanks for this question.
We find some bugs in our released codes for line generation. Besides, we have uploaded a wrong TSR model.
Now both codes and models are updated, and they can work well for the third sample.
![图片 1](https://user-images.githubusercontent.com/12042259/160788113-7628f325-e499-4c73-9159-f6bafa0f63b2.png)
As discussed in the paper, our model is incrementally fine-tuned from LaMa.
Since LaMa has not released weights of discriminator and optimization (until the end of 2021), we retrain it from scratch for 800k steps in Places2 with regular size. There may have some differences, such as masking strategies and learning steps, or even randomness. 
Therefore, the performance of our method can not be ensured to outperform LaMa in any situation (e.g. row1,2).

Update: LaMa has released their weight of big model's discriminator(https://disk.yandex.ru/d/EgqaSnLohjuzAg), our method can also be fine-tuned with it now.
